File: app/src/main/java/com/example/msdc/api/MovieDetails.java
package com.example.msdc.api;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class MovieDetails {
    @SerializedName("title")
    String title;

    @SerializedName("homepage")
    String homepage;

    @SerializedName("overview")
    String overview;

    @SerializedName("runtime")
    String runtime;

    @SerializedName("poster_path")
    String posterPath;

    @SerializedName("vote_average")
    String voteAverage;

    @SerializedName("release_date")
    String releaseDate;

    @SerializedName("original_language")
    String language;

    @SerializedName("backdrop_path")
    String backdropPath;

    @SerializedName("status")
    String status;

    @SerializedName("budget")
    String budget;

    @SerializedName("revenue")
    String revenue;

    @SerializedName("popularity")
    String popularity;

    @SerializedName("tagline")
    String tagline;

    @SerializedName("vote_count")
    String voteCount;

    @SerializedName("genres")
    private final List<GenreResult> genres = new ArrayList<>();

    public String getTitle() {
        return title;
    }

    public String getHomepage() {
        return homepage;
    }

    public String getOverview() {
        return overview;
    }

    public String getRuntime() {
        return runtime;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public String getVoteAverage() {
        return voteAverage;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public String getLanguage() {
        return language;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public String getStatus() {
        return status;
    }

    public String getBudget() {
        return budget;
    }

    public String getRevenue() {
        return revenue;
    }

    public String getPopularity() {
        return popularity;
    }

    public String getTagline() {
        return tagline;
    }

    public String getVoteCount() {
        return voteCount;
    }

    public List<GenreResult> getGenres() {
        return genres;
    }
}

File: app/src/main/java/com/example/msdc/api/MovieDetails.kt
package com.example.msdc.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class MovieDetails (
    @field:SerializedName("title") val title: String? = null,
    @field:SerializedName("homepage") val homepage: String? = null,
    @field:SerializedName("overview") val overview: String? = null,
    @field:SerializedName("runtime") val runtime: String? = null,
    @field:SerializedName("poster_path") val posterPath: String? = null,
    @field:SerializedName("vote_average") val voteAverage: String? = null,
    @field:SerializedName("release_date") val releaseDate: String? = null,
    @field:SerializedName("original_language") val language: String? = null,
    @field:SerializedName("backdrop_path") val backdropPath: String? = null,
    @field:SerializedName("status") val status: String? = null,
    @field:SerializedName("budget") val budget: String? = null,
    @field:SerializedName("revenue") val revenue: String? = null,
    @field:SerializedName("popularity") val popularity: String? = null,
    @field:SerializedName("tagline") val tagline: String? = null,
    @field:SerializedName("vote_count") val voteCount: String? = null,
    @field:SerializedName("genres") val genres: ArrayList<GenreResult>? = null
) : Parcelable